src/bn256/fq.rs

```diff
@@ -257,7 +267,12 @@ impl SqrtRatio for Fq {
     const T_MINUS1_OVER2: [u64; 4] = [0, 0, 0, 0];
 
     fn get_lower_32(&self) -> u32 {
+        #[cfg(any(not(feature = "asm"), not(target_arch = "x86_64")))]
         let tmp = Fq::montgomery_reduce(self.0[0], self.0[1], self.0[2], self.0[3], 0, 0, 0, 0);
+
+        #[cfg(all(feature = "asm", target_arch = "x86_64"))]
```

Comment:
I noticed that this configuration means that if someone enables the `asm` feature in a non-`x86_64` architecture, the `asm` field arithmetic implementation will not be enabled, which may be confusing for the user.  So I think it would be ideal that when the `asm` feature is enabled, if the architecture is not supported there's a compilation error (maybe through a `build.rs`?)

Reply:
Make sense, let me try to use `build.rs` to throw an error.

Reply:
Add in https://github.com/privacy-scaling-explorations/halo2curves/pull/4/commits/3c6f9b90ee7d96caa653ff3ef6180aa9363cedbd!